src/sagas/bucket.js

```diff
@@ -133,35 +136,41 @@ export function* deleteCollection(getState, action) {
   }
 }
 
-export function* listBucketCollections(getState, action) {
+export function* listBucketCollections(getState: GetStateFn, action: Object): SagaGen {
   const {bid} = action;
   try {
     const bucket = getBucket(bid);
-    const {data} = yield call([bucket, bucket.listCollections]);
-    yield put(listBucketCollectionsSuccess(data));
+    const result = yield call([bucket, bucket.listCollections]);
+    if (result == null) {
+      throw new Error("Empty result.");
+    }
+    yield put(listBucketCollectionsSuccess(result.data));
   } catch(error) {
     yield put(notifyError("Couldn't list bucket collections.", error));
   }
 }
 
-export function* listBucketHistory(getState, action) {
+export function* listBucketHistory(getState: GetStateFn, action: Object): SagaGen {
   const {bid, filters: {resource_name, since}} = action;
   try {
     const bucket = getBucket(bid);
-    const {data} = yield call([bucket, bucket.listHistory], {
+    const result = yield call([bucket, bucket.listHistory], {
       since,
       filters: {
         resource_name,
         exclude_resource_name: "record"
       }
     });
-    yield put(listBucketHistorySuccess(data));
+    if (result == null) {
```

Comment:
Hmm flow v0.36 doesn't raise these errors anymore, so I've removed the checks https://github.com/Kinto/kinto-admin/pull/343/commits/43e1ee0744b12c5e3d6640fbb3eb7d530e0c3f12

it may be due to this bugfix which has landed in 0.34 (we were using 0.33 previously) https://github.com/facebook/flow/issues/2589

I'm stil lfeeling there's too much cargo cult ongoing here, but oh well.